bindings/src/main/java/io/pravega/storage/hdfs/HDFSStorage.java

```diff
@@ -511,30 +510,42 @@ public SegmentHandle create(String streamSegmentName) throws StreamSegmentExcept
                     });
         } catch (Exception e) {
             log.error("Exception occurred while listing the segments.", e);
+            throw e;
         }
-        return Collections.emptyIterator();
     }
 
     /**
      * Iterator for segments in HDFS Storage.
      */
     public static class HDFSSegmentIterator implements Iterator<SegmentProperties> {
```

Comment:
Same comments here:
1. Make private.
2. Move to the bottom of the file
3. I don't think you can use `stream()` here since you are already working with an iterator. Use the `Iterators` class. There's a `filter` method that you can use.